File: Medium/2026-03-25-3546-Equal-Sum-Grid-Partition-I/solution.java
class Solution {
    public boolean canPartitionGrid(int[][] grid) {
        int m = grid.length, n = grid[0].length;
        int totalSum = 0;

        // Calculate total sum of the grid
        for (int[] row : grid) {
            for (int num : row) {
                totalSum += num;
            }
        }

        // Check for horizontal cuts
        int rowSum = 0;
        for (int i = 0; i < m - 1; i++) {
            for (int num : grid[i]) {
                rowSum += num;
            }
            if (rowSum * 2 == totalSum) {
                return true;
            }
        }

        // Check for vertical cuts
        int[] colSums = new int[n];
        for (int j = 0; j < n; j++) {
            for (int i = 0; i < m; i++) {
                colSums[j] += grid[i][j];
            }
        }

        int colSum = 0;
        for (int j = 0; j < n - 1; j++) {
            colSum += colSums[j];
            if (colSum * 2 == totalSum) {
                return true;
            }
        }

        return false;
    }
}

File: Medium/2026-03-25-3546-Equal-Sum-Grid-Partition-I/solution.js
var canPartition = function(grid) {
    const m = grid.length;
    const n = grid[0].length;
    
    // Calculate total sum of the grid
    let totalSum = 0;
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            totalSum += grid[i][j];
        }
    }
    
    // Check for horizontal cuts
    let rowSum = 0;
    for (let i = 0; i < m - 1; i++) {
        rowSum += grid[i].reduce((a, b) => a + b, 0);
        if (rowSum * 2 === totalSum) {
            return true;
        }
    }
    
    // Check for vertical cuts
    let colSum = new Array(n).fill(0);
    for (let j = 0; j < n; j++) {
        for (let i = 0; i < m; i++) {
            colSum[j] += grid[i][j];
        }
    }
    
    let currentColSum = 0;
    for (let j = 0; j < n - 1; j++) {
        currentColSum += colSum[j];
        if (currentColSum * 2 === totalSum) {
            return true;
        }
    }
    
    return false;
};